### ISSUE TITLE
Type auto registration during scheme build

### ISSUE THREAD

**Issue by stalniy** [prisma/nexus#136]
I’m building dynamic GraphQL scheme. 

The issue is that in some cases I can’t know the type before hand, so can't pass it into `types` option of `makeSchema` function but I can build it lazily during scheme building. However there is no way to register a new type during scheme build process, also nexus doesn’t register unknown types found on `field.type` as `graphql-js` does.

**Expected behavior**:
nexus should register types found in `field.type` automatically (similar to graphql-js)

**Nexus example**:

It's very simplified just to show the issue
https://codesandbox.io/s/rjloyj3q4n

It throws 
```
/*
 * Error: Looks like you forgot to import MyQuery
 *        in the root "types" passed to Nexus makeSchema
 */
```

**graphql-js Example**

The same example works correctly if using `graphql-js`: https://codesandbox.io/s/ojpv91ojl6

**Comment by stalniy** [comment]
I can try to make a PR if you guys approve this behavior

**Comment by unmyke** [comment]
+1
It's very annoying to pass scalars, enums, interfaces, types to the "types" prop of config object of makeSchema in order based on dependencies of types on each other. It would be more convenient if I can pass only "root" types and import into it other types, which root type depends (don't need to hardcode string in "type" prop of t.field, only pass imported type to it).

**Comment by tgriesser** [comment]
Yeah I agree this could be useful, going to take a look at how tricky it might be to add.

**Comment by tgriesser** [comment]
#143 should solve this - checkout the code there and the kitchen sink example, and try it out using the newly published `0.12.0-beta.1`.
